### PR TITLE
Travis speedup

### PR DESCRIPTION
**Pull request by ulope**
- ~Uses `eth-tester` during 'regular' Travis runs (PR, pushes, etc.).~ (Only infrastructure for now, the switch will happen once the test failures with tester are fixed)
- `Geth` is only tested during a once a day cron run
- Splits linting out into a separate build stage so failures terminate quicker

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -48,16 +48,27 @@ install:
...
   - secure: IXBsyQtH29Vkh+Pe2exrbE3L8FJMQFqJ8ZRxkACts7cQtB8Iz1vyjWg9nYE9ZuCj/JWEeMZd/09JvwwKUj8ZEzwj59gFwVQFwTAxJbiDLRsn7WpdI5Q2fQ9ZPZIAbPo/mJejeHC+z3d5UgY72hbhqWuPJAa4ApWWKE5mPFUIr9uxgs01ReWs/y5HaPawQkSQAKVWWsS5R52Oyr9CYQNbfqfWcoLvzdiIZpsBi2r4ZK3NGrBZPGo4b+PkDkWjuBhMJ0FVABFCJT/bT2ORFsmsCDwZ4I3vOrKtJGDybmwONZqr0ymfYo1lbcUp0mE0zJ0ApyRtLqEFiTzaQqenlAZmBAtpDZVvpxFuDwZgFxafpNutO3Aj3Xbfe+aaooPfHA7SoxmxG/3gWY+OyaME8EDePfBHM0c1gGsNHmbPLt8k0lmwYKlNTFtFFyRAbL3700j19utkGroOK6CUYbed9YD96UehQTj7HN8rpLTZzSMh39c1JHVyqxsUZKkhQgY4GPgx2RAIiCVrwc6wN3Ebtwft0hA2UhvDodsc/qBAyz/YnSp2oKZKagLy5747torZybtNOGKCaV2fT3mSTxV2UNwPJ/N94dlTquJNx3StHT0IqD3Kfo5HYKJKHeri6lttTDul3rjAs1xxB2aAMutsyg7dRbBMmuKlK9gAtoS3UKthQdk=
+stages:
+  - lint
+  - test
+  - deploy
 jobs:
   include:
+    - stage: lint
+      sudo: false
+      before_install: /bin/true
```
Comment (palango) [inline .travis.yml]: why are these necessary? Do we need to overwrite the defaults?

Comment (ulope) [reply .travis.yml]: Yeah, unfortunately the matrix configuration bleeds into all other stages :(

**Comment by hackaugusto** [comment]
Is there any way to use a different `.travis.yaml` for the nigthly? Because we only need `sudo: true` for the tests using geth

File: .travis/install.sh [hunk 2]
```diff
@@ -9,11 +9,10 @@ if [[ "${TRAVIS_OS_NAME}" == "osx" ]]; then
     INSTALL_OPT="--user"
 fi
 
-pip install ${INSTALL_OPT} -U pip wheel coveralls "coverage<4.4"
+pip install ${INSTALL_OPT} --upgrade pip wheel coveralls "coverage<4.4"
```
Comment (palango) [inline .travis/install.sh]: Do we need `--upgrade-strategy eager` here as well?

**Comment by ulope** [comment]
> Is there any way to use a different .travis.yaml for the nigthly?

@hackaugusto 
No, unfortunately not.  We could do it if we split the matrix into individual jobs but that would be quite verbose.
Travis really have maneuvered themselves into a corner syntax-wise...
